pyhive/sqlalchemy_hive.py

```diff
@@ -517,6 +518,37 @@ def _check_unicode_description(self, connection):
         # We decode everything as UTF-8
         return True
 
+
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.expression import Select
+
+
+@compiles(Select)
+def select_star_if_all_columns(select, compiler, **kwargs):
+    froms = select.froms
+    select_columns = list(select.columns.keys())
+    from_columns = list(chain.from_iterable((f.columns.keys() for f in froms)))
```

Comment:
Does this work if the FROM clause has fancy stuff like JOINs or subqueries? I'm pretty sure those require MR jobs.

It might be safest/easiest to whitelist to queries of the form `SELECT col1, col2, ... FROM single_table [LIMIT x]`. I'm pretty sure anything more than that involving WHERE,